### ISSUE TITLE
Document Guidance for Scaling out with Redis as a Backplane

### ISSUE THREAD

**Issue by jgaver** [aspnet/SignalR#3193]
I am currently exploring the proper way to deploy redis as the backplane for several asp.net core based SignalR applications. High Availability is a requirement, but it is unclear what is supported from the redis side.

Things I have tried:

1. Use a Redis cluster, configure SignalR connection string to be aware of all nodes
1.1. When monitoring pub/sub for each node, messages are spread around  nodes within the cluster (it was asymmetric, that is, every 8th message would consistently go to one node while many more would go to another) 
1.1. When killing a node in the cluster using `DEBUG SEGFAULT`, traffic did seem to route to the remaining nodes
2. Use Redis in a Master-Slave Configuration with 3 sentinels for HA
2.1. When monitoring pub/sub for each node, messages seem to be sent to all nodes whether they are the master or a slave
2.2. When killing a node with `DEBUG SEGFAULT`, some messages are missed but eventually things seem to stabilize after sentinel elects a new master
2.3. When simulating bad responsiveness with `DEBUG SLEEP 60`, sentinel will elect a new master but the connected SignalR applications will just hang until the sleep period finishes

Because we are only interested in redis for pub/sub, a full-blown cluster seems like overkill so I'd prefer to go with a Master/Slave approach. The Asp.Net version of SignalR did not support clustering at all, but I cannot find documentation on whether the Core version does.

To sum up, what is the guidance to achieve High Availability while using redis as the backplane? Is Redis clustering supported? What are the pros-cons of clustering vs. Master-Slave with Sentinel for high availability?

**Comment by muratg** [comment]
We're covering some of the doc work here: https://github.com/aspnet/Docs/issues/7814. Please provide any requests there.

**Comment by muratg** [comment]
cc @tdykstra

**Comment by anurse** [comment]
We don't have SignalR-specific guidance around Redis high-availability. I'd suggest looking at some Redis-specific guidance here. SignalR only uses pub/sub so you don't really need any 'durability' guarantees around your data, you just need to make sure all the servers get all the messages.

**Comment by jgaver** [comment]
Thanks @anurse. Do you provide any guidance around how to properly deploy redis as the backplane? How are other users deploying this? 

As an example of the type of information I am looking for - the ASP.NET version had documentation explicitly saying that clustering was not supported. I can find no such guidance for SignalR Core. If this information was available, I could rule out clustering as an option. 

As it stands now I am forced to create prototypes and try to reverse engineer how SignalR is operating, and my results might miss important edge cases that your development team knows about.

Thanks in advance!

**Comment by anurse** [comment]
We don't officially support clustering (as you noted, there is very limited documentation on Redis right now, this is being worked on in aspnet/Docs#7814). I don't know for sure that it doesn't work, but we don't generally test in that scenario.
